ui/v2.5/src/components/Shared/StashBoxIDSearchModal.tsx

```diff
@@ -0,0 +1,445 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Form, Button, Row, Col, Badge, InputGroup } from "react-bootstrap";
+import { FormattedMessage, useIntl } from "react-intl";
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import * as GQL from "src/core/generated-graphql";
+import { ModalComponent } from "src/components/Shared/Modal";
+import { LoadingIndicator } from "src/components/Shared/LoadingIndicator";
+import { stashboxDisplayName } from "src/utils/stashbox";
+import { TruncatedText } from "src/components/Shared/TruncatedText";
+import TextUtils from "src/utils/text";
+import GenderIcon from "src/components/Performers/GenderIcon";
+import { CountryFlag } from "src/components/Shared/CountryFlag";
+import { Icon } from "src/components/Shared/Icon";
+import {
+  stashBoxPerformerQuery,
+  stashBoxStudioQuery,
+  stashBoxSceneQuery,
+} from "src/core/StashService";
+import { useToast } from "src/hooks/Toast";
+import { stringToGender } from "src/utils/gender";
+
+const CLASSNAME = "StashBoxIDSearchModal";
+const CLASSNAME_LIST = `${CLASSNAME}-list`;
+const CLASSNAME_LIST_CONTAINER = `${CLASSNAME_LIST}-container`;
+
+export type EntityType = "performer" | "studio" | "scene";
+
+interface IProps {
+  entityType: EntityType;
+  stashBoxes: GQL.StashBox[];
+  onHide: () => void;
+  onSelectItem: (stashId: string, endpoint: string) => void;
```

Comment:
The typical pattern is for `onSelect` to return an optional object (in this case probably a `GQL.StashIdInput` object), with a return type of `undefined` indicating that the operation was canceled.